crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -3946,26 +3946,33 @@ impl<'db> TypeInferenceBuilder<'db> {
             &alias.name.id
         };
 
+        // Avoid looking up attributes on a module if a module imports from itself
+        // (e.g. `from parent import submodule` inside the `parent` module).
+        let import_is_self_referential =
+            file_to_module(self.db(), self.file()).is_some_and(|m| m.name() == &module_name);
```

Comment:
I think this could possibly be cheaper and more accurate if we compared the files rather than the module names?

```suggestion
        let import_is_self_referential = module_ty
            .into_module_literal()
            .is_some_and(|module| self.file() == module.module(self.db()).file());
```